alacritty/src/event.rs

```diff
@@ -431,10 +445,8 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
     #[inline]
     fn push_search(&mut self, c: char) {
         if let Some(regex) = self.search_state.regex.as_mut() {
-            // Prevent previous search selections from sticking around when not in vi mode.
-            if !self.terminal.mode().contains(TermMode::VI) {
-                self.terminal.selection = None;
-            }
+            // Clear selection so we do not obstruct any matches.
+            self.terminal.selection = None;
```

Comment:
I've decided to do this in `push_search` since this is where it actually matters. There's no need to clear the selection while it is still empty, there's also the possibility that you have something selected after search started and then you don't see it anymore. 

So doing it here makes sure that whenever you type something new, the selection will never obstruct anything.